### PR TITLE
c8d: Use the same logic to get the present images

### PR DESCRIPTION
**Pull request by rumpl**
**- What I did**

Inspect and history used two different ways to find the present images. This made history fail in some cases where image inspect would work (if a configuration of a manifest wasn't found in the content store).

With this change we now use the same logic for both inspect and history.

Related change for inspect https://github.com/moby/moby/pull/46238

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: daemon/containerd/image_history.go [hunk 1]
```diff
@@ -2,18 +2,14 @@ package containerd
 
 import (
 	"context"
-	"sort"
 
-	"github.com/containerd/containerd/images"
 	containerdimages "github.com/containerd/containerd/images"
```
Comment (thaJeztah) [inline daemon/containerd/image_history.go:6]: LOL. Again!

We _really_ need to look which linter we're missing that should check for this.

File: daemon/containerd/image.go [hunk 2]
```diff
@@ -42,44 +42,10 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options bac
 		platform = platforms.OnlyStrict(*options.Platform)
 	}
 
-	var presentImages []imagespec.DockerOCIImage
-	err = i.walkImageManifests(ctx, desc, func(img *ImageManifest) error {
```
Comment (thaJeztah) [inline daemon/containerd/image.go:46]: I think there's actually a bug in this code; but I found some other code that needs to be looked at as well; see

- https://github.com/moby/moby/pull/47183

File: daemon/containerd/image.go [hunk 3]
```diff
@@ -156,6 +122,56 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options bac
...
+					"manifestDescriptor": img.Target(),
+				}).Debug("manifest was present, but accessing its config failed, ignoring")
+				return nil
+			}
+			return errdefs.System(fmt.Errorf("failed to get config descriptor: %w", err))
+		}
+
+		var ociimage imagespec.DockerOCIImage
+		if err := readConfig(ctx, i.content, conf, &ociimage); err != nil {
+			if cerrdefs.IsNotFound(err) {
```
Comment (vvoland) [inline daemon/containerd/image.go:144]: `readConfig` is our function and uses our errdefs 😛 
```suggestion
			if errdefs.IsNotFound(err) {
```

Comment (rumpl) [reply daemon/containerd/image.go:144]: Oh boy, you are right

**Review by vvoland — APPROVED** [review]
LGTM

**Review by dmcgowan — APPROVED** [review]

File: daemon/containerd/image.go (hunk 2 above)
Comment (rumpl) [reply daemon/containerd/image.go:46]: I added platform filtering. Ptal

**Comment by thaJeztah** [comment]
failures on Windows are expected with c8d snapshotters enabled. Bringing this one in

**Pull request merged**